framework/src/meshgenerators/MeshGenerator.C

```diff
@@ -135,15 +238,52 @@ MeshGenerator::generateInternal()
   return mesh;
 }
 
-std::unique_ptr<MeshBase> &
-MeshGenerator::addMeshSubgenerator(const std::string & generator_name,
+void
+MeshGenerator::addMeshSubgenerator(const std::string & type,
                                    const std::string & name,
-                                   InputParameters & params)
+                                   InputParameters params)
```

Comment:
why not a reference anyomore?

Reply:
Because we modify the params. We do this for other factory-style methods where we don't want to modify something someone else passes in. It's either `const T &` or `T` for a copy